### ISSUE TITLE
UDF with type arrow unexpectedly requires alphabetical ordering of returned struct field 

### ISSUE THREAD

**Issue by chilang** [duckdb/duckdb#10808]
### What happens?

When defining a Python UDF which return a `struct` type and with `type="arrow"`, duckdb requires the struct fields to be defined in alphabetical order. This unexpected and can be quite confusing, when switch from `native` function to `arrow` type function results in an strange `Mismatch Type Error`.

### To Reproduce

For example:
```
import duckdb


def gen_person(r: int):
    return [{"name": "John", "age": 30}]


def gen_person_arr(r: int):
    return [{"name": "John", "age": 30}]


with duckdb.connect() as con:
    return_type = duckdb.struct_type({"name": duckdb.typing.VARCHAR, "age": duckdb.typing.INTEGER})
    con.create_function(
        "gen_person",
        gen_person_arr,
        [duckdb.typing.INTEGER],
        duckdb.struct_type({"name": duckdb.typing.VARCHAR, "age": duckdb.typing.INTEGER}),
        side_effects=True,
    )
    for row in con.execute("SELECT gen_person(1)").fetchall():
        print(row)

    # arrow version fail with
    # duckdb.duckdb.TypeMismatchException: Mismatch Type Error: Type STRUCT(age BIGINT, "name" VARCHAR) does not match with STRUCT("name" VARCHAR, age INTEGER). Cannot cast STRUCTs with different names
    con.create_function(
        "gen_person_arr",
        gen_person_arr,
        [duckdb.typing.INTEGER],
        duckdb.struct_type({"name": duckdb.typing.VARCHAR, "age": duckdb.typing.INTEGER}),
        side_effects=True,
        type="arrow",
    )
    for row in con.execute("SELECT gen_person_arr(1)").fetchall():
        print(row)


```

### OS:

aarch64, x64

### DuckDB Version:

0.9.2, nightly (0.10.1.dev253)

### DuckDB Client:

Python

### Full Name:

Chi Lang Ngo

### Affiliation:

NA

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build and a nightly build.

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Tishj** [comment]
I'm confused

You define both of the udfs with `gen_person_arr` as the python method
You then return a list instead of a single item

The problem is not with the arrow udf, the problem occurs in the native udf, because the type you're returning is not what is expected (struct[] vs struct)

**Comment by chilang** [comment]
Sorry, there was a copy paste error in the original example. Here is the one that fails on 0.9.2
```
import duckdb


def gen_person(r: int):
    return {"name": "John", "age": 30}


def gen_person_arr(r: int):
    return [{"name": "John", "age": 30}]


with duckdb.connect() as con:
    return_type = duckdb.struct_type({"name": duckdb.typing.VARCHAR, "age": duckdb.typing.INTEGER})
    con.create_function(
        "gen_person",
        gen_person,
        [duckdb.typing.INTEGER],
        duckdb.struct_type({"name": duckdb.typing.VARCHAR, "age": duckdb.typing.INTEGER}),
        side_effects=True,
    )
    for row in con.execute("SELECT gen_person(1)").fetchall():
        print(row)

    # arrow version fail with
    # duckdb.duckdb.TypeMismatchException: Mismatch Type Error: Type STRUCT(age BIGINT, "name" VARCHAR) does not match with STRUCT("name" VARCHAR, age INTEGER). Cannot cast STRUCTs with different names
    con.create_function(
        "gen_person_arr",
        gen_person_arr,
        [duckdb.typing.INTEGER],
        duckdb.struct_type({"name": duckdb.typing.VARCHAR, "age": duckdb.typing.INTEGER}),
        side_effects=True,
        type="arrow",
    )
    for row in con.execute("SELECT gen_person_arr(1)").fetchall():
        print(row)
```

but it looks like it's fixed on 0.10.0 and nightly now.
Thanks
